src/frontend/src/handler/create_source.rs

```diff
@@ -288,17 +329,14 @@ pub async fn handle_create_source(
     )
     .await?;
 
+    let watermark_descs =
+        bind_source_watermark(&session, name.clone(), stmt.source_watermarks, &columns)?;
+    // TODO(yuhao): allow multiple watermark on source.
+    assert!(watermark_descs.len() <= 1);
```

Comment:
Returning `Err` might be better?

```suggestion
    ensure!(watermark_descs.len() <= 1);
```

Reply:
This is already ensured in the parser. `watermark_descs.len() > 1` should never happen here.